### DISCUSSION TITLE
key-order for task when before or after loops?

### DISCUSSION THREAD

**Discussion by ssbarnea** [ansible/ansible-lint#2455] [Polls]
When having a task with keys like below, which order is better to use:

```
- name: ...
  action: ...
  with_dict: ...  # <-- is looping better before when or after
  when: ...
  notify: ...
```

That is important in relation to https://github.com/ansible/ansible-lint/pull/2454 which enables us to improve the sorting or task keys above requiring name to be the 1st key.

If you happen to also add an explanation why you would find one better than another, even better as we could improve the documentation.

**Comment by tumbl3w33d** [comment]
I don't mind the order in this case too much, however, it certainly makes a difference with blocks and we started putting the `when` on top (after the `name`) because blocks are inherently longer than task calls and the condition of a block tends to get overlooked or misindented, which is even worse.

**Reply by ssbarnea** [reply]
You are right and we already considered this. The question was between these two as we already went to the conclusion that block should always be the last.

It will take long time till we will establish a reference order, but we will take care of the most important keys first and avoid reordering everything in one go. We only order keys when we have a proof that specific order is better. For others, we will leave it as the user wrote it.

**Reply by bcoca** [reply]
don't center on English grammar, diff cultures will want diff ordering (i have slides from company in which `name:` was at bottom of tasks)

**Comment by cidrblock** [comment]
I prefer after, because the task would read more like a sentence....

"Do this, this many times, only if that"

**Reply by apatard** [reply]
> I prefer after, because the task would read more like a sentence....
> 
> "Do this, this many times, only if that"

Same here. That's why I tend to always put ``when:`` in the end.
